### ISSUE TITLE
Deployment on DigitalOcean cluster exactly as in Getting Started guide fails after some time

### ISSUE THREAD

**Issue by royrijkers** [k8ssandra/k8ssandra#866]
## Bug Report
**Describe the bug**
After following the getting started guide exactly it works for maybe around 15 minutes and then it stops working. It seems to be a storage issue, but I am unable to find what to do next.

These Pods fail:

k8ssandra-dc1-default-sts-0
k8ssandra-reaper

the k8ssandra-reaper shows this error:
```
i.c.ReaperApplication - Storage is not ready yet, trying again to connect shortly... 
java.lang.IllegalArgumentException: Failed to add contact point: k8ssandra-dc1-service
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:943)
	at com.datastax.driver.core.Cluster$Builder.addContactPoints(Cluster.java:976)
	at systems.composable.dropwizard.cassandra.CassandraFactory.build(CassandraFactory.java:470)
	at systems.composable.dropwizard.cassandra.CassandraFactory.build(CassandraFactory.java:447)
	at io.cassandrareaper.storage.CassandraStorage.<init>(CassandraStorage.java:224)
	at io.cassandrareaper.ReaperApplication.initializeStorage(ReaperApplication.java:480)
	at io.cassandrareaper.ReaperApplication.tryInitializeStorage(ReaperApplication.java:303)
	at io.cassandrareaper.ReaperApplication.run(ReaperApplication.java:181)
	at io.cassandrareaper.ReaperApplication.run(ReaperApplication.java:98)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:43)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:87)
	at io.dropwizard.cli.Cli.run(Cli.java:78)
	at io.dropwizard.Application.run(Application.java:93)
	at io.cassandrareaper.ReaperApplication.main(ReaperApplication.java:117)
Caused by: java.net.UnknownHostException: k8ssandra-dc1-service: System error
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:939)
```

k8ssandra-dc1-default-sts-0 shows this error:
```
WARN  [epollEventLoopGroup-16-2] 2021-06-07 15:48:11,080 Loggers.java:39 - [s11] Error connecting to Node(endPoint=/tmp/cassandra.sock, hostId=null, hashCode=538f9853), trying next node (FileNotFoundException: null)
```

**To Reproduce**

1. Create cluster on DigitalOcean with 4 basic nodes  (2.5 GB RAM usable (4 GB Total) / 2 vCPUs per node), I have also tried 6 GB RAM usable (8 GB Total) / 4 vCPUs per node without success 
1. Then create the storage class as in the guide: `kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml
` , I also tried: 
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: standard
parameters:
  type: do-block-storage
provisioner: dobs.csi.digitalocean.com
reclaimPolicy: Delete
allowVolumeExpansion: true
volumeBindingMode: WaitForFirstConsumer
```
1. Then follow the exact steps in the getting started guide with the default config.

**Expected behavior**
That cassandra keeps working

**Environment (please complete the following information):**

* Helm charts version info
```k8ssandra   	default  	1       	2021-06-08 12:22:14.03246 +0200 CEST 	deployed	k8ssandra-1.2.0```

* Helm charts user-supplied values
```USER-SUPPLIED VALUES:
cassandra:
  allowMultipleNodesPerWorker: true
  cassandraLibDirVolume:
    size: 5Gi
    storageClass: standard
  datacenters:
  * name: dc1
    racks:
    * name: default
    size: 1
  heap:
    newGenSize: 1G
    size: 1G
  resources:
    limits:
      cpu: 1000m
      memory: 2Gi
    requests:
      cpu: 1000m
      memory: 2Gi
  version: 3.11.10
kube-prometheus-stack:
  enabled: false
  grafana:
    adminPassword: admin123
    adminUser: admin
stargate:
  cpuLimMillicores: 1000
  cpuReqMillicores: 200
  enabled: true
  heapMB: 256
  replicas: 1
```
* Kubernetes version information:
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-14T14:49:35Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
```
* Kubernetes cluster kind:
digitalocean managed cluster

┆Issue is synchronized with this [Jira Bug](https://k8ssandra.atlassian.net/browse/K8SSAND-554) by [Unito](https://www.unito.io)

**Comment by ds-steven-matison** [comment]
@royrijkers You will need to get a valid storage class for your D.O kubernetes engine.  Looks like it might be "do-block-storage" but you confirm with D.O documentation.   I am a D.O user so I would love for you to share confirmation of the correct storage class.

**Comment by royrijkers** [comment]
@ds-steven-matison Thanks for your reply. I just tried the StorageClass configuration exactly as you suggested but that did have the same result unfortunately.

maybe I can find the needed D.O parameters somewhere

**Comment by bradfordcp** [comment]
```
WARN  [epollEventLoopGroup-16-2] 2021-06-07 15:48:11,080 Loggers.java:39 - [s11] Error connecting to Node(endPoint=/tmp/cassandra.sock, hostId=null, hashCode=538f9853), trying next node (FileNotFoundException: null)
```

is a common error message and just indicates that C* is not running. Can you take a look at the `system-loggger` container? This will have the output from C* attempting to start.

**Comment by ds-steven-matison** [comment]
D.O. docs are as challenged as some of the rest...  i did find this, which indicates maybe we are close:

```
{
  "kind": "StorageClass",
  "apiVersion": "storage.k8s.io/v1",
  "metadata": {
    "name": "do-block-storage",
   [...]
  },
  "provisioner": "dobs.csi.digitalocean.com",
  "reclaimPolicy": "Delete",
  "volumeBindingMode": "Immediate",
  "allowVolumeExpansion": true     <== What you are saying is this will not work
}
```

In the OP they are saying the allow will not work, so do a test iteration without that, and without the parameters.

**Comment by bradfordcp** [comment]
`  "volumeBindingMode": "Immediate",` is a little concerning. We recommend using `WaitForFirstConsumer` on storage classes used.

**Comment by royrijkers** [comment]
@ds-steven-matison Thanks for the suggestion, I tried it but unfortunately without success. 

@bradfordcp These are some logs from the "server-system-logger" container:

```
INFO  [main] 2021-06-08 14:26:05,166 Config.java:537 - Node configuration:[allocate_tokens_for_keyspace=null; authenticator=PasswordAuthenticator; authorizer=CassandraAuthorizer; auto_bootstrap=true; auto_snapshot=true; back_pressure_enabled=false; back_pressure_strategy=null; batch_size_fail_threshold_in_kb=640; batch_size_warn_threshold_in_kb=64; batchlog_replay_throttle_in_kb=1024; broadcast_address=null; broadcast_rpc_address=10.244.1.168; buffer_pool_use_heap_if_exhausted=true; cas_contention_timeout_in_ms=1000; cdc_enabled=false; cdc_free_space_check_interval_ms=250; cdc_raw_directory=null; cdc_total_space_in_mb=0; check_for_duplicate_rows_during_compaction=true; check_for_duplicate_rows_during_reads=true; client_encryption_options=<REDACTED>; cluster_name=k8ssandra; column_index_cache_size_in_kb=2; column_index_size_in_kb=64; commit_failure_policy=stop; commitlog_compression=null; commitlog_directory=null; commitlog_max_compression_buffers_in_pool=3; commitlog_periodic_queue_size=-1; commitlog_segment_size_in_mb=32; commitlog_sync=periodic; commitlog_sync_batch_window_in_ms=NaN; commitlog_sync_period_in_ms=10000; commitlog_total_space_in_mb=null; compaction_large_partition_warning_threshold_mb=100; compaction_throughput_mb_per_sec=16; concurrent_compactors=null; concurrent_counter_writes=32; concurrent_materialized_view_writes=32; concurrent_reads=32; concurrent_replicates=null; concurrent_writes=32; counter_cache_keys_to_save=2147483647; counter_cache_save_period=7200; counter_cache_size_in_mb=null; counter_write_request_timeout_in_ms=5000; credentials_cache_max_entries=1000; credentials_update_interval_in_ms=3600000; credentials_validity_in_ms=3600000; cross_node_timeout=false; data_file_directories=[Ljava.lang.String;@f001896; disk_access_mode=auto; disk_failure_policy=stop; disk_optimization_estimate_percentile=0.95; disk_optimization_page_cross_chance=0.1; disk_optimization_strategy=ssd; dynamic_snitch=true; dynamic_snitch_badness_threshold=0.1; dynamic_snitch_reset_interval_in_ms=600000; dynamic_snitch_update_interval_in_ms=100; enable_materialized_views=true; enable_sasi_indexes=true; enable_scripted_user_defined_functions=false; enable_user_defined_functions=false; enable_user_defined_functions_threads=true; encryption_options=; endpoint_snitch=GossipingPropertyFileSnitch; file_cache_round_up=null; file_cache_size_in_mb=null; gc_log_threshold_in_ms=200; gc_warn_threshold_in_ms=1000; hinted_handoff_disabled_datacenters=[]; hinted_handoff_enabled=true; hinted_handoff_throttle_in_kb=1024; hints_compression=null; hints_directory=null; hints_flush_period_in_ms=10000; incremental_backups=false; index_interval=null; index_summary_capacity_in_mb=null; index_summary_resize_interval_in_minutes=60; initial_token=null; inter_dc_stream_throughput_outbound_megabits_per_sec=200; inter_dc_tcp_nodelay=false; internode_authenticator=null; internode_compression=dc; internode_recv_buff_size_in_bytes=0; internode_send_buff_size_in_bytes=0; key_cache_keys_to_save=2147483647; key_cache_save_period=14400; key_cache_size_in_mb=null; listen_address=10.244.1.168; listen_interface=null; listen_interface_prefer_ipv6=false; listen_on_broadcast_address=false; max_hint_window_in_ms=10800000; max_hints_delivery_threads=2; max_hints_file_size_in_mb=128; max_mutation_size_in_kb=null; max_streaming_retries=3; max_value_size_in_mb=256; memtable_allocation_type=heap_buffers; memtable_cleanup_threshold=null; memtable_flush_writers=0; memtable_heap_space_in_mb=null; memtable_offheap_space_in_mb=null; min_free_space_per_drive_in_mb=50; native_transport_flush_in_batches_legacy=true; native_transport_max_concurrent_connections=-1; native_transport_max_concurrent_connections_per_ip=-1; native_transport_max_concurrent_requests_in_bytes=-1; native_transport_max_concurrent_requests_in_bytes_per_ip=-1; native_transport_max_frame_size_in_mb=256; native_transport_max_negotiable_protocol_version=-2147483648; native_transport_max_threads=128; native_transport_port=9042; native_transport_port_ssl=null; num_tokens=256; otc_backlog_expiration_interval_ms=200; otc_coalescing_enough_coalesced_messages=8; otc_coalescing_strategy=DISABLED; otc_coalescing_window_us=200; partitioner=org.apache.cassandra.dht.Murmur3Partitioner; permissions_cache_max_entries=1000; permissions_update_interval_in_ms=3600000; permissions_validity_in_ms=3600000; phi_convict_threshold=8.0; prepared_statements_cache_size_mb=null; range_request_timeout_in_ms=10000; read_request_timeout_in_ms=5000; repair_session_max_tree_depth=18; request_scheduler=org.apache.cassandra.scheduler.NoScheduler; request_scheduler_id=null; request_scheduler_options=null; request_timeout_in_ms=10000; role_manager=CassandraRoleManager; roles_cache_max_entries=1000; roles_update_interval_in_ms=3600000; roles_validity_in_ms=3600000; row_cache_class_name=org.apache.cassandra.cache.OHCProvider; row_cache_keys_to_save=2147483647; row_cache_save_period=0; row_cache_size_in_mb=0; rpc_address=0.0.0.0; rpc_interface=null; rpc_interface_prefer_ipv6=false; rpc_keepalive=true; rpc_listen_backlog=50; rpc_max_threads=2147483647; rpc_min_threads=16; rpc_port=9160; rpc_recv_buff_size_in_bytes=null; rpc_send_buff_size_in_bytes=null; rpc_server_type=sync; saved_caches_directory=null; seed_provider=org.apache.cassandra.locator.K8SeedProvider{seeds=k8ssandra-seed-service}; server_encryption_options=; slow_query_log_timeout_in_ms=500; snapshot_before_compaction=false; snapshot_on_duplicate_row_detection=false; ssl_storage_port=7001; sstable_preemptive_open_interval_in_mb=50; start_native_transport=true; start_rpc=true; storage_port=7000; stream_throughput_outbound_megabits_per_sec=200; streaming_keep_alive_period_in_secs=300; streaming_socket_timeout_in_ms=86400000; thrift_framed_transport_size_in_mb=15; thrift_max_message_length_in_mb=16; thrift_prepared_statements_cache_size_mb=null; tombstone_failure_threshold=100000; tombstone_warn_threshold=1000; tracetype_query_ttl=86400; tracetype_repair_ttl=604800; transparent_data_encryption_options=org.apache.cassandra.config.TransparentDataEncryptionOptions@13f17eb4; trickle_fsync=true; trickle_fsync_interval_in_kb=10240; truncate_request_timeout_in_ms=60000; unlogged_batch_across_partitions_warn_threshold=10; user_defined_function_fail_timeout=1500; user_defined_function_warn_timeout=500; user_function_timeout_policy=die; windows_timer_interval=1; write_request_timeout_in_ms=2000]
INFO  [main] 2021-06-08 14:26:05,167 DatabaseDescriptor.java:381 - DiskAccessMode 'auto' determined to be mmap, indexAccessMode is mmap
INFO  [main] 2021-06-08 14:26:05,168 DatabaseDescriptor.java:439 - Global memtable on-heap threshold is enabled at 256MB
INFO  [main] 2021-06-08 14:26:05,168 DatabaseDescriptor.java:443 - Global memtable off-heap threshold is enabled at 256MB
WARN  [main] 2021-06-08 14:26:05,206 DatabaseDescriptor.java:503 - Small commitlog volume detected at /opt/cassandra/data/commitlog; setting commitlog_total_space_in_mb to 1243.  You can override this in cassandra.yaml
WARN  [main] 2021-06-08 14:26:05,209 DatabaseDescriptor.java:530 - Small cdc volume detected at /opt/cassandra/data/cdc_raw; setting cdc_total_space_in_mb to 621.  You can override this in cassandra.yaml
WARN  [main] 2021-06-08 14:26:06,588 DatabaseDescriptor.java:579 - Only 4.831GiB free across all data volumes. Consider adding more capacity to your cluster or removing obsolete snapshots
INFO  [main] 2021-06-08 14:26:06,706 RateBasedBackPressure.java:123 - Initialized back-pressure with high ratio: 0.9, factor: 5, flow: FAST, window size: 2000.
INFO  [main] 2021-06-08 14:26:06,708 DatabaseDescriptor.java:775 - Back-pressure is disabled with strategy null.
INFO  [main] 2021-06-08 14:26:07,862 GossipingPropertyFileSnitch.java:64 - Loaded cassandra-topology.properties for compatibility
WARN  [main] 2021-06-08 14:26:07,972 K8SeedProvider3x.java:54 - Seed provider couldn't lookup host k8ssandra-seed-service
ERROR [main] 2021-06-08 14:26:07,974 CassandraDaemon.java:803 - Exception encountered during startup: The seed provider lists no seeds.
```

Maybe I should increase the size from 5Gib to 20Gib or something

**Comment by bradfordcp** [comment]
The disk size is fine (although you cannot change it easily later). The issue here is that it couldn't find a seed host. Since you're creating a single node cluster it _should_ itself be a seed. `GossipingPropertyFileSnitch.java:64 - Loaded cassandra-topology.properties for compatibility` is concerning and should be dealt with, but that's not a blocker here (@jdonenine do we have a ticket for removing `cassandra-topology.properties`?)

Can you share any labels that are on this pod?

**Comment by royrijkers** [comment]
@bradfordcp 

These labels are on the `k8ssandra-dc1-default-sts-0` pod:
```
    app.kubernetes.io/managed-by: cass-operator
    cassandra.datastax.com/cluster: k8ssandra
    cassandra.datastax.com/datacenter: dc1
    cassandra.datastax.com/node-state: Started-not-Ready
    cassandra.datastax.com/rack: default
    controller-revision-hash: k8ssandra-dc1-default-sts-bfb4dd97d
    statefulset.kubernetes.io/pod-name: k8ssandra-dc1-default-sts-0
```

Do you happen to know how I make it a seed itself?

**Comment by bradfordcp** [comment]
The operator _should_ handle that. I'm spinning up a DO environment to take a look at this.

**Comment by bradfordcp** [comment]
When you have a minute @royrijkers check out https://github.com/k8ssandra/k8ssandra/pull/868. It's a full install guide for DigitalOcean Kubernetes. I was able to get K8ssandra deployed along with Medusa with backups going to DigitalOcean Spaces. There is a sample `values.yaml` file at https://github.com/k8ssandra/k8ssandra/pull/868/files#diff-52458c0d7612d706b5e6449350df54e58e1e0d468a8dca479c2ad0d5fbe9c3b8 that I used in my environment. You will want to adjust the resources and heap to match the resources available in your droplets. Please let me know if this helps you get past the issue you were seeing.

**Comment by royrijkers** [comment]
@bradfordcp Thank you so much for writing this document.

I tried to do it but taints did not allow me to install it.

More information here: https://github.com/k8ssandra/k8ssandra/pull/868/files#r648044294

**Comment by royrijkers** [comment]
@bradfordcp Thanks for writing the guide

I have tried to follow the guide except for tainting the node pool because I am just using 1 node-pool at the moment with 4 basic nodes.

It runs into the same "seed" problem

```
INFO  [main] 2021-06-09 08:44:43,647 DatabaseDescriptor.java:381 - DiskAccessMode 'auto' determined to be mmap, indexAccessMode is mmap
INFO  [main] 2021-06-09 08:44:43,647 DatabaseDescriptor.java:439 - Global memtable on-heap threshold is enabled at 256MB
INFO  [main] 2021-06-09 08:44:43,648 DatabaseDescriptor.java:443 - Global memtable off-heap threshold is enabled at 256MB
WARN  [main] 2021-06-09 08:44:43,710 DatabaseDescriptor.java:503 - Small commitlog volume detected at /opt/cassandra/data/commitlog; setting commitlog_total_space_in_mb to 1243.  You can override this in cassandra.yaml
WARN  [main] 2021-06-09 08:44:43,712 DatabaseDescriptor.java:530 - Small cdc volume detected at /opt/cassandra/data/cdc_raw; setting cdc_total_space_in_mb to 621.  You can override this in cassandra.yaml
WARN  [main] 2021-06-09 08:44:44,649 DatabaseDescriptor.java:579 - Only 4.825GiB free across all data volumes. Consider adding more capacity to your cluster or removing obsolete snapshots
INFO  [main] 2021-06-09 08:44:44,746 RateBasedBackPressure.java:123 - Initialized back-pressure with high ratio: 0.9, factor: 5, flow: FAST, window size: 2000.
INFO  [main] 2021-06-09 08:44:44,747 DatabaseDescriptor.java:775 - Back-pressure is disabled with strategy null.
INFO  [main] 2021-06-09 08:44:45,335 GossipingPropertyFileSnitch.java:64 - Loaded cassandra-topology.properties for compatibility
WARN  [main] 2021-06-09 08:44:45,421 K8SeedProvider3x.java:54 - Seed provider couldn't lookup host k8ssandra-seed-service
ERROR [main] 2021-06-09 08:44:45,424 CassandraDaemon.java:803 - Exception encountered during startup: The seed provider lists no seeds.
```

config file:

```
cassandra:
  version: "3.11.10"
  cassandraLibDirVolume:
    storageClass: do-block-storage
    size: 5Gi
  allowMultipleNodesPerWorker: true
  heap:
    size: 1G
    newGenSize: 1G
  resources:
    requests:
      cpu: 1000m
      memory: 2Gi
    limits:
      cpu: 1000m
      memory: 2Gi
  datacenters:
    - name: dc1
      size: 1
      racks:
        - name: default
  tolerations:
    - key: "app"
      operator: "Equal"
      value: "cassandra"
      effect: "NoSchedule"
kube-prometheus-stack:
  grafana:
    adminUser: admin
    adminPassword: admin123
  # Tolerations for Cassandra nodes
stargate:
  enabled: true
  replicas: 1
  heapMB: 256
  cpuReqMillicores: 200
  cpuLimMillicores: 1000
medusa:
  enabled: true
  storage: s3_compatible
  storage_properties:
    host: ams3.digitaloceanspaces.com
    port: 443
    secure: "True"
  bucketName: db-backups
  storageSecret: prod-k8ssandra-medusa-key

```

Storage class:

```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: do-block-storage
provisioner: dobs.csi.digitalocean.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
allowVolumeExpansion: true
```

**Comment by bradfordcp** [comment]
Please correct me if I am wrong, but it sounds like the cluster came up, went down, then for some reason failed to come back up as there were no seeds. Is that accurate? After running the `helm install` command does the `...dc1-default-sts-0` pod ever show as `3/3`?

@jsanda @burmanm I wonder what situations could cause a pod to lost the seed label. Unless the pod is destroyed, recreated by the statefulset controller and we never reapply the label. Is that possible?

**Comment by royrijkers** [comment]
@bradfordcp 

In the config I used a "size" of 1 where I specified the datacenters. 

At some point it showed 1/1 under `k8ssandra-dc1-default-sts-0`

I can try to have 3 as a size, maybe that solves the problem

**Comment by royrijkers** [comment]
Oh, now I see that the "size" is probably the data rack size and not the "replica" size.
When I set the size to 3 it did spin up 3 replicas of the k8ssandra-dc1-default-sts-0 pod though

**Comment by royrijkers** [comment]
@bradfordcp Will have a look to see if I see 3/3

**Comment by jsanda** [comment]
> Please correct me if I am wrong, but it sounds like the cluster came up, went down, then for some reason failed to come back up as there were no seeds. Is that accurate? After running the `helm install` command does the `...dc1-default-sts-0` pod ever show as `3/3`?
> 
> @jsanda @burmanm I wonder what situations could cause a pod to lost the seed label. Unless the pod is destroyed, recreated by the statefulset controller and we never reapply the label. Is that possible?

I need to review the code to be certain but I am inclined to say cass-operator will remove the seed label from a pod if it does not become ready after some period of time.

**Comment by jsanda** [comment]
> The disk size is fine (although you cannot change it easily later). The issue here is that it couldn't find a seed host. Since you're creating a single node cluster it _should_ itself be a seed. `GossipingPropertyFileSnitch.java:64 - Loaded cassandra-topology.properties for compatibility` is concerning and should be dealt with, but that's not a blocker here (@jdonenine do we have a ticket for removing `cassandra-topology.properties`?)
> 
> Can you share any labels that are on this pod?

We need a ticket for this AFAIK.

**Comment by royrijkers** [comment]
> A size of 1 should still work. I run K8ssandra in that configuration locally. 1/1 may display while the pod is initializing (we have an initContainer) but if you have Medusa enabled we should see 3/3 when it's fully up and responding to healthchecks.

After 10 minutes it still shows 1/1. it seems to run fine now, but i am quite sure the seed problem will come up again.

It seems to respond to health checks:

```
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:18:16,794 Cli.java:617 - address=/10.244.1.202:32770 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:18:26,799 Cli.java:617 - address=/10.244.1.202:32778 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:18:31,294 Cli.java:617 - address=/10.244.1.202:32782 url=/api/v0/probes/liveness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:18:36,794 Cli.java:617 - address=/10.244.1.202:32788 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:18:46,287 Cli.java:617 - address=/10.244.1.202:32796 url=/api/v0/probes/liveness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:18:46,795 Cli.java:617 - address=/10.244.1.202:32798 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:18:56,794 Cli.java:617 - address=/10.244.1.202:32806 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:19:01,287 Cli.java:617 - address=/10.244.1.202:32810 url=/api/v0/probes/liveness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:19:06,793 Cli.java:617 - address=/10.244.1.202:32816 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:19:16,287 Cli.java:617 - address=/10.244.1.202:32824 url=/api/v0/probes/liveness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:19:16,794 Cli.java:617 - address=/10.244.1.202:32826 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:19:26,795 Cli.java:617 - address=/10.244.1.202:32834 url=/api/v0/probes/readiness status=200 OK
INFO  [nioEventLoopGroup-2-1] 2021-06-09 15:19:31,287 Cli.java:617 - address=/10.244.1.202:32838 url=/api/v0/probes/liveness status=200 OK
INFO  [nioEventLoopGroup-2-2] 2021-06-09 15:19:36,798 Cli.java:617 - address=/10.244.1.202:32846 url=/api/v0/probes/readiness status=200 OK
```

**Comment by royrijkers** [comment]
k8ssandra-dc1-default-sts-0 is now running for 14 hours and it still shows 1/1

**Comment by royrijkers** [comment]
Now it crashed again and it has no seed label anymore

**Comment by bradfordcp** [comment]
How much vCPU / RAM is available per worker? I'm curious what is causing the crash (in addition to the seed problem).

**Comment by royrijkers** [comment]
@bradfordcp 

the cluster itself is made with 4 x Basic node (2.5 GB RAM usable (4 GB Total) / 2 vCPUs per node)

In the k8ssandra.yaml file `cpu: 1000m, memory: 2Gi` is specified: 

```json
cassandra:
  version: "3.11.10"
  cassandraLibDirVolume:
    storageClass: do-block-storage
    size: 5Gi
  allowMultipleNodesPerWorker: true
  heap:
    size: 1G
    newGenSize: 1G
  resources:
    requests:
      cpu: 1000m
      memory: 2Gi
    limits:
      cpu: 1000m
      memory: 2Gi
  datacenters:
    - name: dc1
      size: 1
      replicas: 3
      racks:
        - name: default
  tolerations:
    - key: "app"
      operator: "Equal"
      value: "cassandra"
      effect: "NoSchedule"
```

**Comment by bradfordcp** [comment]
Sorry about closing the ticket there, we linked it to the install docs PR.

I'm curious if the pods are getting OOM killed. Did you add any schema / tables to the cluster?

**Comment by royrijkers** [comment]
@bradfordcp 

No problem. Yes I did,  and only after that it seems to have this issue

**Comment by royrijkers** [comment]
@bradfordcp I can try if it runs longer with a higher memory limit. But I think that that is a separate issue from the seed label that is missing

**Comment by royrijkers** [comment]
I run into the exact same issue on GKE

config: 

```
cassandra:
  version: "3.11.10"
  cassandraLibDirVolume:
    storageClass: standard-rwo
    size: 5Gi
  allowMultipleNodesPerWorker: true
  heap:
    size: 1G
    newGenSize: 1G
  resources:
    requests:
      cpu: 1000m
      memory: 2Gi
    limits:
      cpu: 1000m
      memory: 2Gi
  datacenters:
    - name: dc1
      size: 1
      replicas: 1
      racks:
        - name: default
  tolerations:
    - key: "app"
      operator: "Equal"
      value: "cassandra"
      effect: "NoSchedule"
kube-prometheus-stack:
  grafana:
    adminUser: admin
    adminPassword: admin123
  # Tolerations for Cassandra nodes
stargate:
  enabled: true
  replicas: 1
  heapMB: 256
  cpuReqMillicores: 200
  cpuLimMillicores: 1000
medusa:
  enabled: true
  storage: s3_compatible
  storage_properties:
    host: ams3.digitaloceanspaces.com
    port: 443
    secure: "True"
  bucketName: streampal-backups
  storageSecret: prod-k8ssandra-medusa-key
```

errors: 

```
INFO  [main] 2021-06-17 08:33:57,491 DatabaseDescriptor.java:381 - DiskAccessMode 'auto' determined to be mmap, indexAccessMode is mmap
INFO  [main] 2021-06-17 08:33:57,491 DatabaseDescriptor.java:439 - Global memtable on-heap threshold is enabled at 256MB
INFO  [main] 2021-06-17 08:33:57,492 DatabaseDescriptor.java:443 - Global memtable off-heap threshold is enabled at 256MB
WARN  [main] 2021-06-17 08:33:57,509 DatabaseDescriptor.java:503 - Small commitlog volume detected at /opt/cassandra/data/commitlog; setting commitlog_total_space_in_mb to 1243.  You can override this in cassandra.yaml
WARN  [main] 2021-06-17 08:33:57,559 DatabaseDescriptor.java:530 - Small cdc volume detected at /opt/cassandra/data/cdc_raw; setting cdc_total_space_in_mb to 621.  You can override this in cassandra.yaml
WARN  [main] 2021-06-17 08:33:58,771 DatabaseDescriptor.java:579 - Only 4.824GiB free across all data volumes. Consider adding more capacity to your cluster or removing obsolete snapshots
INFO  [main] 2021-06-17 08:33:58,886 RateBasedBackPressure.java:123 - Initialized back-pressure with high ratio: 0.9, factor: 5, flow: FAST, window size: 2000.
INFO  [main] 2021-06-17 08:33:58,887 DatabaseDescriptor.java:775 - Back-pressure is disabled with strategy null.
INFO  [main] 2021-06-17 08:33:59,971 GossipingPropertyFileSnitch.java:64 - Loaded cassandra-topology.properties for compatibility
WARN  [main] 2021-06-17 08:34:00,256 K8SeedProvider3x.java:54 - Seed provider couldn't lookup host k8ssandra-seed-service
ERROR [main] 2021-06-17 08:34:00,261 CassandraDaemon.java:803 - Exception encountered during startup: The seed provider lists no seeds.
INFO  [main] 2021-06-17 08:34:23,437 SystemDistributedReplicationInterceptor.java:73 - Using override for distributed system keyspaces: {dc1=1, class=NetworkTopologyStrategy}
INFO  [main] 2021-06-17 08:34:25,397 YamlConfigurationLoader.java:92 - Configuration location: file:/etc/cassandra/cassandra.yaml
```

**Comment by royrijkers** [comment]
@bradfordcp I installed an old chart v0.19.0 and this works fine

**Comment by burmanm** [comment]
Your problem is still the same, you allocate more memory for Cassandra than you allocate to the pod:

```
  heap:
    size: 1G
    newGenSize: 1G
  resources:
    requests:
      cpu: 1000m
      memory: 2Gi
    limits:
      cpu: 1000m
```

heap limit must be lower (like half of the total allocated pod memory resources) for Cassandra to work, since it needs memory for other than heap as well.

**Comment by royrijkers** [comment]
@burmanm Thanks, will try that!

**Comment by johnsmartco** [comment]
Roy, @royrijkers, thanks for your earlier feedback. Were you able to try the revised memory settings? I'll be happy to tweak our docs, and I'm interested in whether Michael @burmanm 's recommendation worked in your DigitalOcean environment. Thanks Roy.

**Comment by johnsmartco** [comment]
Closing this issue following updated heap limit in the documented doks.values.yaml, https://docs.k8ssandra.io/install/doks/#generate-doksvaluesyaml .  
We will revisit this issue if @royrijkers has subsequent feedback w.r.t. any related / remaining issues he encounters with the install of K8ssandra on DigitalOcean. Thanks guys for the observations and ideas above.

**Comment by royrijkers** [comment]
@burmanm After I brought down the heap limit it kept working for many days in stead of minutes but it ran eventually into the same issue. 

There was not a lot of use of the DB and there is not much information stored there yet. But we do use quite a lot of keyspaces.

What is the best way to see where to allocate more or less memory or CPU? 

Also stargate stops working after an update: https://github.com/k8ssandra/k8ssandra/issues/1085 (Not sure if it has anything to do with DOKS)
